### PR TITLE
DEV: Setup experimental sidebar skeleton

### PR DESCRIPTION
**Pull request by tgxworld**
* hidden siteSetting to enable experimental sidebar
* user preference to enable experimental sidebar
* `experimental_sidebar_enabled` attribute for current user
* Empty glimmer component for Sidebar

### CODE REVIEW

**Comment by tgxworld** [comment]
Client side tests have been intentionally left out for enabling the sidebar in the user's preferences menu because this is experimental and is a really low risk code path.

File: app/assets/javascripts/discourse/app/templates/application.hbs [hunk 1]
```diff
@@ -25,6 +25,8 @@
       {{create-topics-notice}}
       {{plugin-outlet name="top-notices" connectorTagName="div" args=(hash currentPath=router._router.currentPath)}}
     </div>
+
+    <Sidebar />
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/templates/application.hbs:29]: Should the `if currentUser.experimental_sidebar_enabled` be outside the component? Happy either way just wondering

File: app/assets/javascripts/discourse/app/components/sidebar.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import GlimmerComponent from "discourse/components/glimmer";
+
+export default class Sidebar extends GlimmerComponent {}
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/sidebar.js:3]: Did you know this component is empty!?! :P (jk I know it is intentionally empty)

**Review by martin-brennan — APPROVED** [review]
Very minor comments.

**Comment by tgxworld** [comment]
Thank you @martin-brennan

**Pull request merged**